src/v3/stylis-logical-plugin/src/__tests__/plugin.ts

```diff
@@ -0,0 +1,221 @@
+import { setupSerializer } from '../testUtils';
+import { ensmallen } from '../utils';
+
+describe('compile, transform, and serialize', () => {
+  const processor = setupSerializer();
+
+  it('handles comments', () => {
+    const css = `
+      /* comment */
+    `;
+
+    expect(processor(css)).toBe(ensmallen(css));
+  });
+
+  it('handles ruleset without logical declaration', () => {
+    const css = `
+      [dir="ltr"] .a {
+        color: red;
+      }
+      [dir="rtl"] .a {
+        color: red;
+      }
+    `;
+
+    expect(processor(css)).toBe(ensmallen(css));
+  });
+
+  it('handles ruleset with simple logical declaration', () => {
+    const css = `
+      .a {
+        margin-inline-end: 5px;
+      }
+    `;
+
+    expect(processor(css)).toBe(ensmallen(`
+      [dir="ltr"] .a {
+        margin-right: 5px;
+      }
+      [dir="rtl"] .a {
+        margin-left: 5px;
+      }
+    `));
+  });
+
+  it('handles ruleset with some declarations that do not need transform', () => {
+    const css = `
+      .a {
+        color: red;
+        margin-inline-end: 5px;
+      }
+    `;
+
+    expect(processor(css)).toBe(ensmallen(`
+      [dir="ltr"] .a {
+        color: red;
+        margin-right: 5px;
+      }
+      [dir="rtl"] .a {
+        color: red;
+        margin-left: 5px;
+      }
+    `));
+  });
+
+  it('handles ruleset with multiple rules', () => {
+    const css = `
+      .a, .b {
+        margin-inline-end: 5px;
+      }
+    `;
+
+    expect(processor(css)).toBe(ensmallen(`
+      [dir="ltr"] .a, [dir="ltr"] .b {
+        margin-right: 5px;
+      }
+      [dir="rtl"] .a, [dir="rtl"] .b {
+        margin-left: 5px;
+      }
+    `));
+  });
+
+  it('handles ruleset with mixture of declarations', () => {
+    const css = `
+      .a {
+        color: red;
+        padding-inline-end: 5px;
+        padding-block-end: 5px;
+      }
+    `;
+
+    expect(processor(css)).toBe(ensmallen(`
+      [dir="ltr"] .a {
+        color: red;
+        padding-right: 5px;
+        padding-bottom: 5px;
+      }
+      [dir="rtl"] .a {
+        color: red;
+        padding-left: 5px;
+        padding-bottom: 5px;
+      }
+    `));
+  });
```

Comment:
@lesterchoi-okta see latest commit f4c0adb -- I was able to get this working using a `:not` pseudo-class since we can assume LTR and we have a unique "root" element containing the top-level `dir=` attribute.

Reply:
Luckily the `:not` pseudo-class is supported in IE11 😄